core/src/test/java/org/testcontainers/utility/AuthenticatedImagePullTest.java

```diff
@@ -153,6 +154,13 @@ public void testThatAuthLocatorIsUsedForDockerComposePull() throws IOException {
         }
     }
 
+    @NotNull
+    private Path getLocalTempDir() throws IOException {
+        Path projectRoot = Paths.get(".");
+        Path tempDirectory = Files.createTempDirectory(projectRoot, this.getClass().getSimpleName() + "-test-");
```

Comment:
From Javadoc:
```java
     * <p> As with the {@code createTempFile} methods, this method is only
     * part of a temporary-file facility. A {@link Runtime#addShutdownHook
     * shutdown-hook}, or the {@link java.io.File#deleteOnExit} mechanism may be
     * used to delete the directory automatically.
```

Let's use `deleteOnExit` to avoid the leftovers

Reply:
Bleurgh, of course. Thanks.

Reply:
Replaced with an implementation that deletes both at exit.